src/Serializers/QueryResultSerializer.php

```diff
@@ -56,6 +56,39 @@ public function isSerializerFor( $queryResult ) {
 	}
 
 	/**
+	 * Get display title of a page.
+	 *
+	 * @param int $pageId page ID of page being queried
+	 *
+	 * @return string|bool display title value or false if not found
+	 */
+	public static function getDisplayTitle( $pageId ) {
+
+		if ( ! is_numeric( $pageId ) || $pageId < 1 ) {
+
+			return false;
+
+		}
+
+		$dbr = wfGetDB( DB_SLAVE );
```

Comment:
The PageProps class at [0] encapsulates the DB interactions and adds caching. Would it suffice to add that class (or a modification thereof) to SMW at this point, since the earliest it would be included in MW core would be MW 1.27? I also plan to pursue the discussion of whether the display title functionality in core would be better served being part of a core class such as Article or WikiPage rather than as a page property. So, the mechanism for accessing the display title *may* require a change when MW 1.27 is released, but there would not be a change from the perspective of the user, and it would be highly beneficial to have this capability sooner rather than later.

[0] https://gerrit.wikimedia.org/r/#/c/246246/